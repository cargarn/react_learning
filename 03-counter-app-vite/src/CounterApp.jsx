import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState(value);
    const handleAdd = () => {
        setCounter(counter + 1);
        //setCounter( (c) => c - 1 );
    }
    const handleSubtract = () => {
        setCounter(counter - 1);
    }
    const handleReset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }>+</button>
            <button onClick={ handleSubtract }>-</button>
            <button onClick={ handleReset }>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}