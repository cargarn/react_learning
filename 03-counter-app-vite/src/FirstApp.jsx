import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};


FirstApp.defaultProps = { // Entran antes que los propTypes definidos mas abajo
  title: "No hay titulo",
  subTitle: "No hay subtitulo",
  name: "Carlos Garnacho"
};