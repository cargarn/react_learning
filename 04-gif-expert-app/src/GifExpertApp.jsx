import React from 'react';
import { useState } from 'react';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState();

  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}

      {/* Listado de Gifs */}
        {/* Gif Item */}
    </>
  )
}
