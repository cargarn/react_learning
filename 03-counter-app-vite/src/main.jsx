import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
//import { FirstApp } from './FirstApp'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {
      //<FirstApp title="titulo desde main.jsx" subTitle="wow, soy subtitulo" />
    }
    <CounterApp value={0} />
  </React.StrictMode>
)
