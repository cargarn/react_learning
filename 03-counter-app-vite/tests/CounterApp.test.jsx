import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {

    const initialValue = 10;

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <CounterApp value={ initialValue } /> );
        expect(container).toMatchSnapshot();
    });

    test('debe de encontrar el valor inicial pasado por props', () => {
        render( <CounterApp value={ 100 } /> );
        expect( screen.getByText(100) ).toBeTruthy();
    });

    test('debe de incrementar con el boton +1', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click(screen.getByText("+"));
        expect( screen.getByText(initialValue + 1) ).toBeTruthy();
    });
    
    test('debe de incrementar con el boton -1', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click(screen.getByText("-"));
        expect( screen.getByText(initialValue - 1) ).toBeTruthy();
    });
        
    test('debe de funcionar el boton de reset', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click(screen.getByText("+")); // incrementamos el valor
        fireEvent.click(screen.getByText("Reset")); // reseteamos el valor inicial
        expect( screen.getByText(initialValue) ).toBeTruthy();
    });
});