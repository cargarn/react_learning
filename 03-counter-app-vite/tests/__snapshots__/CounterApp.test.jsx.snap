// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +
  </button>
  <button>
    -
  </button>
  <button>
    Reset
  </button>
</div>
`;
